exchanges/binance.js

```diff
@@ -32,8 +32,8 @@ var recoverableErrors = new RegExp(
 );
 
 Trader.prototype.retry = function(method, args, error) {
-  if (!error || !error.message.match(recoverableErrors)) {
-    log.error('[binance.js] ', this.name, 'returned an irrecoverable error');
+  if (!error || (error && !error.msg.match(recoverableErrors))) {
```

Comment:
If `error` isn't a falsy then it's a truthy and there's no requirement to re-check. 
A check for `error.msg` existence before the regex would be beneficial though.

```js
if (!error || (error.msg && !error.msg.match(recoverableErrors))) {
```

Reply:
actually retry() seems not always to be called with defined error, thats why there was "type error undefined for .match" before